python/pyspark/sql/functions/builtin.py

```diff
@@ -15534,19 +15532,7 @@ def to_csv(col: "ColumnOrName", options: Optional[Dict[str, str]] = None) -> Col
     |      2,Alice|
     +-------------+
 
-    Example 2: Converting a complex StructType to a CSV string
-
-    >>> from pyspark.sql import Row, functions as sf
-    >>> data = [(1, Row(age=2, name='Alice', scores=[100, 200, 300]))]
-    >>> df = spark.createDataFrame(data, ("key", "value"))
-    >>> df.select(sf.to_csv(df.value)).show(truncate=False) # doctest: +SKIP
-    +-----------------------+
-    |to_csv(value)          |
-    +-----------------------+
-    |2,Alice,"[100,200,300]"|
```

Comment:
This is a breaking change in fact. We need to update the SQL migration guide at least.

Reply:
+1

Reply:
Let me update.

Reply:
I have updated the document `sql-migration-guide.md`.

Reply:
> I think we should simply disallow this. We could have a legacy conf, but it doesn't much make sense. What does it return for ArrayType and MapType?

@HyukjinKwon See what it does return for `ArrayType` in the example. I doubt that we should simply disallow it w/o any legacy config at least.

Reply:
In `this case` of Python, it is because the data type is ArrayData (specifically implemented as `GenericArrayData`),
This case just goes into the following logic:

https://github.com/apache/spark/blob/09bdf2a9334f210ece4c23d0f3324f81113330d0/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityGenerator.scala#L91-L93

https://github.com/apache/spark/blob/09bdf2a9334f210ece4c23d0f3324f81113330d0/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/GenericArrayData.scala#L85


And it just so happens that `GenericArrayData` overrides the `toString` method, so it appears to display the result `correctly`, perhaps it's just a `coincidence`?